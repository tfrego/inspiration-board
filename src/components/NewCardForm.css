/* The Modal (background) */

.new-card-form {
  width: 50%;
  margin: auto;
  padding-bottom: 4rem;
}

.new-card-form__header {
  text-align: center;
  text-transform: uppercase;
  font-size: 1.5em;
}

.new-card-form__form {
  font-size: 1.25em;
  display: grid;
  grid-template-columns: [labels] auto [controls] 1fr;
  grid-auto-flow: row;
  grid-gap: .8em;
}

.new-card-form__form-label  {
  grid-column: labels;
  grid-row: auto;
}

.new-card-form__form-select,
.new-card-form__form-textarea,
.new-card-form__form-button {
  font-family: 'Raleway', sans-serif;
  font-size: 1.25em;
  grid-column: controls;
  grid-row: auto;
}

.new-card-form__form-button {
  background-color: inherit;
  border: 1px solid black;
  font-size: 1em;
}
