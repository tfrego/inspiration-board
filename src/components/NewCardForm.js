import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';
import './NewCardForm.css';

const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog"]

class NewCardForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      text: '',
      emoji: '',
    }
  }

  onFormChange = (event) => {
    const field = event.target.name;
    const value = event.target.value;

    const updatedState = {};

    updatedState[field] = value;

    this.setState(updatedState);
  }

  onSubmit = (event) => {
    event.preventDefault();
    const { text, emoji } = this.state;

    if (text === '' && emoji === '') return;

    this.props.addCardCallback(this.state);

    this.setState({
      text: '',
      emoji: '',
    })
  }

  render() {
    const getEmoji = (item) => (item && emoji.getUnicode(item));

    return (
      <section className="new-card-form">
        <h3 className="new-card-form__header">Add an inspiration</h3>
        <form className="new-card-form__form" onSubmit={this.onSubmit}>

            <label className="new-card-form__form-label" htmlFor="text">Text</label>
            <textarea className="new-card-form__form-textarea" name="text" onChange={this.onFormChange} value={this.state.text}></textarea>

            <label className="new-card-form__form-label" htmlFor="emoji">Emoji</label>
            <select className="new-card-form__form-select" name="emoji"
              onChange={this.onFormChange} value={this.state.emoji}>
            {EMOJI_LIST.map((value, index) => <option key={index} value={value}>{getEmoji(value)}</option>)}
            </select>

          <input className="new-card-form__form-button" type="submit" name="submit" value="Submit" />
        </form>
      </section>
    );
  }
}

NewCardForm.propTypes = {
  addCardCallback: PropTypes.func.isRequired,
}

export default NewCardForm;
