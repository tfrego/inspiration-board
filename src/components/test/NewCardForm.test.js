import React from 'react';
import NewCardForm from '../NewCardForm';
import { shallow } from 'enzyme';

describe('new card form', () => {
  it('will match the NewCardForm Snapshot', () => {
    const wrapper = shallow( <NewCardForm addCardCallback={() => {} } />);

    expect(wrapper).toMatchSnapshot();
  });
});
