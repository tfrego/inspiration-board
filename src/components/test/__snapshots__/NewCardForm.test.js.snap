// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new card form will match the NewCardForm Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewCardForm
    addCardCallback={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="new-card-form__header"
        >
          Add an inspiration
        </h3>,
        <form
          className="new-card-form__form"
          onSubmit={[Function]}
        >
          <label
            className="new-card-form__form-label"
            htmlFor="text"
          >
            Text
          </label>
          <textarea
            className="new-card-form__form-textarea"
            name="text"
            onChange={[Function]}
            value=""
          />
          <label
            className="new-card-form__form-label"
            htmlFor="emoji"
          >
            Emoji
          </label>
          <select
            className="new-card-form__form-select"
            name="emoji"
            onChange={[Function]}
            value=""
          >
            <option
              value=""
            >
              
            </option>
            <option
              value="heart_eyes"
            >
              😍
            </option>
            <option
              value="beer"
            >
              🍺
            </option>
            <option
              value="clap"
            >
              👏
            </option>
            <option
              value="sparkling_heart"
            >
              💖
            </option>
            <option
              value="heart_eyes_cat"
            >
              😻
            </option>
            <option
              value="dog"
            >
              🐶
            </option>
          </select>
          <input
            className="new-card-form__form-button"
            name="submit"
            type="submit"
            value="Submit"
          />
        </form>,
      ],
      "className": "new-card-form",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add an inspiration",
          "className": "new-card-form__header",
        },
        "ref": null,
        "rendered": "Add an inspiration",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="new-card-form__form-label"
              htmlFor="text"
            >
              Text
            </label>,
            <textarea
              className="new-card-form__form-textarea"
              name="text"
              onChange={[Function]}
              value=""
            />,
            <label
              className="new-card-form__form-label"
              htmlFor="emoji"
            >
              Emoji
            </label>,
            <select
              className="new-card-form__form-select"
              name="emoji"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              >
                
              </option>
              <option
                value="heart_eyes"
              >
                😍
              </option>
              <option
                value="beer"
              >
                🍺
              </option>
              <option
                value="clap"
              >
                👏
              </option>
              <option
                value="sparkling_heart"
              >
                💖
              </option>
              <option
                value="heart_eyes_cat"
              >
                😻
              </option>
              <option
                value="dog"
              >
                🐶
              </option>
            </select>,
            <input
              className="new-card-form__form-button"
              name="submit"
              type="submit"
              value="Submit"
            />,
          ],
          "className": "new-card-form__form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Text",
              "className": "new-card-form__form-label",
              "htmlFor": "text",
            },
            "ref": null,
            "rendered": "Text",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "new-card-form__form-textarea",
              "name": "text",
              "onChange": [Function],
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Emoji",
              "className": "new-card-form__form-label",
              "htmlFor": "emoji",
            },
            "ref": null,
            "rendered": "Emoji",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  value=""
                >
                  
                </option>,
                <option
                  value="heart_eyes"
                >
                  😍
                </option>,
                <option
                  value="beer"
                >
                  🍺
                </option>,
                <option
                  value="clap"
                >
                  👏
                </option>,
                <option
                  value="sparkling_heart"
                >
                  💖
                </option>,
                <option
                  value="heart_eyes_cat"
                >
                  😻
                </option>,
                <option
                  value="dog"
                >
                  🐶
                </option>,
              ],
              "className": "new-card-form__form-select",
              "name": "emoji",
              "onChange": [Function],
              "value": "",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": "",
                  "value": "",
                },
                "ref": null,
                "rendered": "",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": "😍",
                  "value": "heart_eyes",
                },
                "ref": null,
                "rendered": "😍",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "host",
                "props": Object {
                  "children": "🍺",
                  "value": "beer",
                },
                "ref": null,
                "rendered": "🍺",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "3",
                "nodeType": "host",
                "props": Object {
                  "children": "👏",
                  "value": "clap",
                },
                "ref": null,
                "rendered": "👏",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "4",
                "nodeType": "host",
                "props": Object {
                  "children": "💖",
                  "value": "sparkling_heart",
                },
                "ref": null,
                "rendered": "💖",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "5",
                "nodeType": "host",
                "props": Object {
                  "children": "😻",
                  "value": "heart_eyes_cat",
                },
                "ref": null,
                "rendered": "😻",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "6",
                "nodeType": "host",
                "props": Object {
                  "children": "🐶",
                  "value": "dog",
                },
                "ref": null,
                "rendered": "🐶",
                "type": "option",
              },
            ],
            "type": "select",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "new-card-form__form-button",
              "name": "submit",
              "type": "submit",
              "value": "Submit",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="new-card-form__header"
          >
            Add an inspiration
          </h3>,
          <form
            className="new-card-form__form"
            onSubmit={[Function]}
          >
            <label
              className="new-card-form__form-label"
              htmlFor="text"
            >
              Text
            </label>
            <textarea
              className="new-card-form__form-textarea"
              name="text"
              onChange={[Function]}
              value=""
            />
            <label
              className="new-card-form__form-label"
              htmlFor="emoji"
            >
              Emoji
            </label>
            <select
              className="new-card-form__form-select"
              name="emoji"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              >
                
              </option>
              <option
                value="heart_eyes"
              >
                😍
              </option>
              <option
                value="beer"
              >
                🍺
              </option>
              <option
                value="clap"
              >
                👏
              </option>
              <option
                value="sparkling_heart"
              >
                💖
              </option>
              <option
                value="heart_eyes_cat"
              >
                😻
              </option>
              <option
                value="dog"
              >
                🐶
              </option>
            </select>
            <input
              className="new-card-form__form-button"
              name="submit"
              type="submit"
              value="Submit"
            />
          </form>,
        ],
        "className": "new-card-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add an inspiration",
            "className": "new-card-form__header",
          },
          "ref": null,
          "rendered": "Add an inspiration",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="new-card-form__form-label"
                htmlFor="text"
              >
                Text
              </label>,
              <textarea
                className="new-card-form__form-textarea"
                name="text"
                onChange={[Function]}
                value=""
              />,
              <label
                className="new-card-form__form-label"
                htmlFor="emoji"
              >
                Emoji
              </label>,
              <select
                className="new-card-form__form-select"
                name="emoji"
                onChange={[Function]}
                value=""
              >
                <option
                  value=""
                >
                  
                </option>
                <option
                  value="heart_eyes"
                >
                  😍
                </option>
                <option
                  value="beer"
                >
                  🍺
                </option>
                <option
                  value="clap"
                >
                  👏
                </option>
                <option
                  value="sparkling_heart"
                >
                  💖
                </option>
                <option
                  value="heart_eyes_cat"
                >
                  😻
                </option>
                <option
                  value="dog"
                >
                  🐶
                </option>
              </select>,
              <input
                className="new-card-form__form-button"
                name="submit"
                type="submit"
                value="Submit"
              />,
            ],
            "className": "new-card-form__form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Text",
                "className": "new-card-form__form-label",
                "htmlFor": "text",
              },
              "ref": null,
              "rendered": "Text",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "new-card-form__form-textarea",
                "name": "text",
                "onChange": [Function],
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Emoji",
                "className": "new-card-form__form-label",
                "htmlFor": "emoji",
              },
              "ref": null,
              "rendered": "Emoji",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    value=""
                  >
                    
                  </option>,
                  <option
                    value="heart_eyes"
                  >
                    😍
                  </option>,
                  <option
                    value="beer"
                  >
                    🍺
                  </option>,
                  <option
                    value="clap"
                  >
                    👏
                  </option>,
                  <option
                    value="sparkling_heart"
                  >
                    💖
                  </option>,
                  <option
                    value="heart_eyes_cat"
                  >
                    😻
                  </option>,
                  <option
                    value="dog"
                  >
                    🐶
                  </option>,
                ],
                "className": "new-card-form__form-select",
                "name": "emoji",
                "onChange": [Function],
                "value": "",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": "",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": "",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": "😍",
                    "value": "heart_eyes",
                  },
                  "ref": null,
                  "rendered": "😍",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "host",
                  "props": Object {
                    "children": "🍺",
                    "value": "beer",
                  },
                  "ref": null,
                  "rendered": "🍺",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "3",
                  "nodeType": "host",
                  "props": Object {
                    "children": "👏",
                    "value": "clap",
                  },
                  "ref": null,
                  "rendered": "👏",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "4",
                  "nodeType": "host",
                  "props": Object {
                    "children": "💖",
                    "value": "sparkling_heart",
                  },
                  "ref": null,
                  "rendered": "💖",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "5",
                  "nodeType": "host",
                  "props": Object {
                    "children": "😻",
                    "value": "heart_eyes_cat",
                  },
                  "ref": null,
                  "rendered": "😻",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "6",
                  "nodeType": "host",
                  "props": Object {
                    "children": "🐶",
                    "value": "dog",
                  },
                  "ref": null,
                  "rendered": "🐶",
                  "type": "option",
                },
              ],
              "type": "select",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "new-card-form__form-button",
                "name": "submit",
                "type": "submit",
                "value": "Submit",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
